PMMain/Midas/MidasGUI/SharedCommandData.cpp

```diff
@@ -219,4 +308,28 @@ void SharedCommandData::extractPoint(boost::any value)
         point velocity = boost::any_cast<point>(value);
         setVelocity(velocity);
     }
-}
\ No newline at end of file
+}
+
+
+void SharedCommandData::extractOrientation(boost::any value)
+{
+    if (value.type() != typeid(orientation_data))
+    {
+        Filter::setFilterError(filterError::INVALID_INPUT);
+        Filter::setFilterStatus(filterStatus::FILTER_ERROR);
+    }
+    else
+    {
+        orientation_data orientation = boost::any_cast<orientation_data> (value);
+        setMyoOrientation(orientation);
+    }
+}
+
+void SharedCommandData::extractKeySelectAngle(int angle)
+{
+    setKeySelectAngle(angle);
+}
+
```

Comment:
There seems to be conflicting/overlapping logic with keyselectangle and myoOrientation. which one is supposed to do what?